packages/grafana-ui/src/components/Link/TextLink.tsx

```diff
@@ -0,0 +1,118 @@
+import { css } from '@emotion/css';
+import React, { AnchorHTMLAttributes, forwardRef } from 'react';
+
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import { IconName } from '../../types';
+import { Icon } from '../Icon/Icon';
+import { customColor, customWeight } from '../Text/utils';
+
+import { Link } from './Link';
+
+export interface Props extends AnchorHTMLAttributes<HTMLAnchorElement> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'] | 'error' | 'success' | 'warning' | 'info';
+  /** Specify if the link will redirect users to a page in or out Grafana */
+  external?: boolean;
+  /** True if the element should be displayed inline with surrounding text, false if it should be displayed as a block */
+  inline?: boolean;
+  /** What typograpy variant should be used for the component. Only use if default variant for the defined element is not what is needed */
+  variant?: keyof ThemeTypographyVariantTypes;
+  /** Override the default weight for the used variant */
+  weight?: 'light' | 'regular' | 'medium' | 'bold';
+  /** When it is a standalone element, whether to align the text to left, center or right of its block*/
+  textAlignment?: 'left' | 'right' | 'center';
+  /** When the link is inline it won't show an icon.
+   * If it is a standalone element, it will show the icon specified. When this is an external link, the default icon will be 'external-link-alt' */
+  icon?: IconName;
+  children: string;
+}
+
+export const TextLink = forwardRef<HTMLAnchorElement, Props>(
+  ({ href, color = 'link', external = false, inline, variant, weight, icon, children, ...rest }, ref) => {
+    const validUrl = locationUtil.stripBaseFromUrl(textUtil.sanitizeUrl(href ?? ''));
+
+    const theme = useTheme2();
+    const styles = getLinkStyles(theme, variant, weight, color, inline);
+    const externalIcon = icon || 'external-link-alt';
+
+    return external ? (
+      <a href={validUrl} ref={ref} target="_blank" rel="noreferrer" {...rest} className={styles}>
```

Comment:
Is `{...rest}` in the right place? Won't it will allow users of this component to be able to change the `target` and `rel` if `external` is also passed in? And if `external` isn't passed in but `target` and `rel` are we fall into the falsy condition where we then lose the external icon. 🤔 

I get that it means users can treat it like an `<a />` tag but it makes me wonder if having `rest` is a good idea for component APIs.

Reply:
Hi @jackw! First of all, thanks for your comment. We have modified the code and currently omit `target` and `rel` from the `TextLinkProps` so users won't be able to change their values.

Reply:
I'm not sure it is enough to exclude them from the types. I guess it depends if we want to prevent them from being set or not.

In that case, it would probably also be wise to remove them from the object being created by the `...rest` syntax.

Reply:
Hi @mckn,
We've checked and when we tried to add `target` or `rel` to the `TextLink` component we got a squiggly red line and a message saying that that property does not exist, as you can see here:

![Captura de pantalla 2023-07-13 a las 12 45 07](https://github.com/grafana/grafana/assets/65417731/2e74f439-d17f-414e-ac16-7fd812781a33)
![Captura de pantalla 2023-07-13 a las 12 44 39](https://github.com/grafana/grafana/assets/65417731/4b158246-36cf-4be9-9eba-c009ccc1a620)

Reply:
Yeah, but that is only the types. You could still ignore the types, which is not used during runtime, and pass the properties to the component. Not super important but you should be aware of that.

Reply:
Thanks @mckn! I created a [PR](https://github.com/grafana/grafana/pull/71934) to be sure `target` value is always `_blank` and `rel` is always `noreferrer`.